hunting-buddy.lic

```diff
@@ -391,8 +531,13 @@ before_dying do
     stop_script(script_name) if Script.running?(script_name)
   end
   DRCA.release_cyclics(get_settings.cyclic_no_release)
-  Flags.delete('room-check')
-  Flags.delete('familiar_drag')
+  flags_to_delete = [
+    'hunting-buddy-room-check', # there are multiple flags with this prefix, one per friend
+    'hunting-buddy-familiar-drag'
+  ]
+  Flags.flags.keys
```

Comment:
This feels a bit over-engineered, if you compare it to the two lines on the left of the diff.

Reply:
Note that multiple dynamic flags get created for each buddy/friend in your yaml and the original "two lines" did not delete them all. This update ensures all the hunting buddy flags are deleted. Lingering flags consume resources because they are always trying to match incoming text to their patterns.

Reply:
Right understood, I missed that. Where are the buddy flags created?

Reply:
Looks like here? https://github.com/rpherbig/dr-scripts/pull/5526/files#diff-a6df990badd4d65a33f8e58f3337c06a688d7e8921a6029832f5fc79fbaf58d0R299

Reply:
> Looks like here? https://github.com/rpherbig/dr-scripts/pull/5526/files#diff-a6df990badd4d65a33f8e58f3337c06a688d7e8921a6029832f5fc79fbaf58d0R299

Yep, that's it! Yeah, I just stumbled upon it one day randomly checking what flags I had and noticed some residue so set out to clean 'em up. I say "because of performance" but it's probably 99% my OCD and 1% performance" =)

Reply:
> This feels a bit over-engineered, if you compare it to the two lines on the left of the diff.

And good to question these things, I've been known to over-engineer things and it helps to have a fresh pair of eyes review the code and solution approach.

Thanks

Reply:
Closing the loop, the deletion of the flags has been simplified as follows:
```ruby
  Flags.flags.keys
    .select { |flag_name| flag_name.start_with?('hunting-buddy-') }
    .each   { |flag_name| Flags.delete(flag_name) }
```